components/text_normalization/Dockerfile

```diff
@@ -1,18 +1,27 @@
-FROM --platform=linux/amd64 python:3.8-slim
+FROM --platform=linux/amd64 python:3.8-slim as base
 
 ## System dependencies
 RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install git -y
 
+RUN pip3 install fondant[aws,azure,gcp]@git+https://github.com/mrchtr/fondant
+
 # install requirements
 COPY requirements.txt /
 RUN pip3 install --no-cache-dir -r requirements.txt
 
-# Set the working directory to the component folder
-WORKDIR /component/src
-
 # Copy over src-files
-COPY src/ .
+COPY src/ src/
 
+# Run component tests
+FROM base as test
+RUN pip3 install pytest # TODO add pytest to package setup
+ENV PYTHONPATH "${PYTHONPATH}:./src"
+COPY tests/ tests/
+RUN ["python", "-m", "pytest", "tests/"]
+
+FROM base
```

Comment:
Should we name this stage as well so we can explicitly invoke it?

```
FROM base as build
```

If we then call

```
docker build --stage=build
```

It should skip the test stage when using buildkit.

Reply:
We could do it. But we have to make sure to change the build script. Otherwise the next release will fail. 

If we keep it for now as it is, there would be no pressure to update all components immediately. We can still run `docker build .` and only the default stage (without the test stage) is executed.

Reply:
> We can still run docker build . and only the default stage (without the test stage) is executed.

Ok wasn't aware of this, then this is fine. Let's validate that tests do not run on every build to be sure though.